crates/sparrow-merge/src/gather.rs

```diff
@@ -0,0 +1,339 @@
+mod gathered_batches;
+
+use crate::gather::gathered_batches::GatheredBatches;
```

Comment:
should this be `pub use gathered_batches::GatheredBatches`? It seems like things return `GatheredBatches`, so it is part of the API.